docs/references/nextjs/clerk-middleware.mdx

```diff
@@ -0,0 +1,239 @@
+---
+title: clerkMiddleware() | Next.js
+description: The `clerkMiddleware()` function allows you to protect your Next.js application using middleware.
+---
+
+# `clerkMiddleware()`
+
+The `clerkMiddleware()` helper integrates Clerk authentication into your Next.js application through middleware. `clerkMiddleware()` is compatible with both the App and Pages routers. 
+
+## Usage
+
+Create a `middleware.ts` file. If your application uses the `src/` directory, your `middleware.ts` file should be placed inside the `src/` folder. If you are not using a `src/` directory, place the `middleware.ts` file in your root directory. 
+
+<Callout level="info">
+For more information about middleware in Next.js, see the [Next.js documentation](https://nextjs.org/docs/pages/building-your-application/routing/middleware).
+</Callout>
+
+```tsx filename="middleware.ts"
+import { clerkMiddleware } from '@clerk/nextjs/server';
+
+export default clerkMiddleware()
+
+export const config = {
+  matcher: ['/((?!.*\\..*|_next).*)', '/', '/(api|trpc)(.*)'],
+};
+````
+
+This basic Middleware will not protect any routes. The new `clerkMiddleware()` behaves in the opposite manner from the previous `authMiddleware()` that Clerk shipped with version 4 of our SDKs. With `clerkMiddleware()` all routes are now public and you must opt-in to protection for routes.
```

Comment:
My opinion, let's not refer to `clerkMiddleware` as new, that makes the docs time sensitive. Additionally any reference to older behaviors can be mentioned and handled in migration v4 to v5 documentation. 
```suggestion
By default `clerkMiddleware` will not protect any routes. With `clerkMiddleware()` all routes are now public and you must opt-in to protection for routes.
```